File: Medium/3812-smallest-palindromic-rearrangement-i/smallest-palindromic-rearrangement-i.java
class Solution{
    public String smallestPalindrome(String s){
        int n=s.length();

        int[] freq=new int[26];
        for(char ch: s.toCharArray()){
            freq[ch-'a']++;
        }

        char[] temp=new char[n];
        int left=0, right=n-1;
        for(int i=0; i<26 && left<=right; i++){
            while(freq[i]>=2){
                temp[left++]=(char)(i+'a');
                temp[right--]=(char)(i+'a');
                freq[i]-=2;
            }
            if(freq[i] == 1){
                temp[n/2]=(char)(i+'a');
                freq[i]=0;
            }
        }

        return new String(temp);


    }
}

File: 3812-smallest-palindromic-rearrangement-i/smallest-palindromic-rearrangement-i.py
class Solution:
    def smallestPalindrome(self, s: str) -> str:
        n=len(s)

        freq=[0]*26
        for ch in s:
            freq[ord(ch)-ord('a')]+=1
            

        res=['']*n
        left, right=0, n-1
        i=0;
        while i<26 and left<=right:
            while freq[i]>=2:
                res[left]=(chr)(i+ord('a'))
                res[right]=(chr)(i+ord('a'))
                freq[i]-=2
                left += 1
                right -= 1
            
            if freq[i] == 1:
                res[n//2]=(chr)(i+ord('a'))
                freq[i]=0
            
            i+=1
    

        return ''.join(res)